ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -270,16 +286,48 @@ protected static void endToEndTest(Port targetPort, String routerDatacenter, Str
       // verify user metadata
       ByteBuffer userMetadataOutput = MessageFormatRecord.deserializeUserMetadata(responseStream);
       Assert.assertArrayEquals(usermetadata, userMetadataOutput.array());
+      if (testEncryption) {
+        assertNotNull("MessageMetadata should not have been null",
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+        assertArrayEquals("EncryptionKey mismatch", encryptionKey,
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0).getEncryptionKey().array());
+      } else {
+        assertNull("MessageMetadata should have been null",
+            resp3.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+      }
+
+      // get blob all
+      GetRequest getRequest4 =
+          new GetRequest(1, "clientid2", MessageFormatFlags.All, partitionRequestInfoList, GetOption.None);
+      channel.send(getRequest4);
+      stream = channel.receive().getInputStream();
+      GetResponse resp4 = GetResponse.readFrom(new DataInputStream(stream), clusterMap);
+      responseStream = resp4.getInputStream();
+      BlobAll blobAll = MessageFormatRecord.deserializeBlobAll(responseStream, blobIdFactory);
+      byte[] actualBlobData = new byte[(int) blobAll.getBlobData().getSize()];
+      blobAll.getBlobData().getStream().getByteBuffer().get(actualBlobData);
+      // verify content
+      Assert.assertArrayEquals("Content mismatch", data, actualBlobData);
+      if (testEncryption) {
+        Assert.assertNotNull("EncryptionKey should not ne null", blobAll.getBlobEncryptionKey());
+        Assert.assertArrayEquals("EncryptionKey mismatch", encryptionKey, blobAll.getBlobEncryptionKey().array());
+      } else {
+        Assert.assertNull("EncryptionKey should have been null", blobAll.getBlobEncryptionKey());
+      }
 
       // get blob data
-      // Use router to get the blob
-      Properties routerProperties = getRouterProps(routerDatacenter);
-      routerProperties.putAll(routerProps);
-      VerifiableProperties routerVerifiableProps = new VerifiableProperties(routerProperties);
-      Router router = new NonBlockingRouterFactory(routerVerifiableProps, clusterMap, new MockNotificationSystem(9),
-          getSSLFactoryIfRequired(routerVerifiableProps)).getRouter();
-      checkBlobId(router, blobId1, data);
-      router.close();
+      // if encryption is enabled, router will try to decrypt the blob content using encryptionKey which will fail, as
+      // encryptionKey in this test doesn't have any relation to the content. Both are random bytes for test purposes.
```

Comment:
I don't get this comment - why would decrypting the content using the encryption key fail?

Reply:
We directly contact the server here and does not go thru router.